goaviatrix/version.go

```diff
@@ -83,21 +83,24 @@ func (c *Client) AsyncUpgrade(version *Version, upgradeGateways bool) error {
 	if err != nil {
 		return fmt.Errorf("HTTP POST %s failed: %v", form["action"], err)
 	}
+	var data struct {
+		Return bool `json:"return"`
+		Result int  `json:"results"`
+	}
 	buf := new(bytes.Buffer)
 	buf.ReadFrom(resp.Body)
-	requestID, err := strconv.Atoi(buf.String())
-	if err != nil {
+	bodyString := buf.String()
+	bodyIoCopy := strings.NewReader(bodyString)
+	if err = json.NewDecoder(bodyIoCopy).Decode(&data); err != nil {
+		return errors.New("Json Decode " + form["action"] + " failed: " + err.Error() + "\n Body: " + bodyString)
+	}
+
+	requestID := data.Result
+	if requestID == 0 {
```

Comment:
Should we check the Return field in the response?

Reply:
Yes, we should.